lib/iris/tests/unit/fileformats/ff/test__ff_equivalents.py

```diff
@@ -0,0 +1,255 @@
+                                    row_dependent_constants,
+                                    real_constants,
+                                    horiz_grid_type)]
+        return self
+
+    def test__basic(self):
+        # Call with four unnamed args.
+        self.check_call(['cdc', 'rdc', 'rc', 'ht'], {},
+                        expected_result=('cdc', 'rdc', 'rc', 'ht'))
+
+    def test__all_named_keys(self):
+        # Call with four named keys.
+        kwargs = {'column_dependent_constants': 1,
+                  'row_dependent_constants': 2,
+                  'real_constants': 3,
+                  'horiz_grid_type': 4}
+        self.check_call([], kwargs,
+                        expected_result=(1, 2, 3, 4))
+
+    def test__badargs(self):
+        # Just to make sure we can catch bad arguments.
+        msg = r'takes.*\b5\b.*\b6\b.*given'
+        with self.assertRaisesRegexp(TypeError, msg):
+            self.check_call([1, 2, 3, 4, 5], {},
+                            expected_result=(1, 2, 3, 4, 5))
+
+
+class Test_Grid(Mixin_ConstructorTest, Mixin_Grid_Tests, tests.IrisTest):
+    target_class_name = 'Grid'
+
+    def setUp(self):
+        self.constructor_setup()
+
+
+class Test_ArakawaC(Mixin_ConstructorTest, Mixin_Grid_Tests, tests.IrisTest):
+    target_class_name = 'ArakawaC'
+
+    def setUp(self):
+        self.constructor_setup()
+
+
+class Test_ENDGame(Mixin_ConstructorTest, Mixin_Grid_Tests, tests.IrisTest):
+    target_class_name = 'ENDGame'
+
+    def setUp(self):
+        self.constructor_setup()
+
+
+class Test_NewDynamics(Mixin_ConstructorTest, Mixin_Grid_Tests,
+                       tests.IrisTest):
+    target_class_name = 'NewDynamics'
+
+    def setUp(self):
+        self.constructor_setup()
+
+
+class Test_FFHeader(Mixin_ConstructorTest, tests.IrisTest):
+    target_class_name = 'FFHeader'
+
+    @staticmethod
+    def dummy_constructor_call(self, filename, word_depth=16):
+        # A replacement for the 'real' constructor call in the parent class.
+        # Used to check for correct args and kwargs in the call.
+        # Just record the call arguments in a global, for testing.
+        global constructor_calls_data
+        # It's global because in use, we do not have the right 'self' here.
+        # Note: as we append, it contains a full call history.
+        constructor_calls_data += [(filename, word_depth)]
+        return self
+
+    def setUp(self):
+        self.constructor_setup()
+
+    def test__basic(self):
+        # Call with just a filename.
+        self.check_call(['filename'], {},
+                        expected_result=('filename', 16))
+
+    def test__word_depth(self):
+        # Call with a word-depth.
+        self.check_call(['filename'], {'word_depth': 4},
+                        expected_result=('filename', 4))
+
+    def test__badargs(self):
+        # Just to make sure we can catch bad arguments.
+        msg = r'takes.*\b3\b.*\b6\b.*given'
```

Comment:
This is horrific. Any chance of finding a nicer solution which handles py2 & py3?